src/tests/unittests/measurements/test_scans.py

```diff
@@ -74,6 +74,26 @@ def test_scan1D(self, verbose=0):
         self.assertTrue('hi' in data.metadata)
         gates.close()
 
+    @staticmethod
+    def test_scan1D_no_gates():
+        p = Parameter('p', set_cmd=None)
+        r = VoltageDivider(p, 4)
+        scanjob = scanjob_t({'sweepdata': {'param': p, 'start': 0, 'end': 10, 'step': 2}, 'minstrument': [r]})
+        station=qcodes.Station()
+        _ = scan1D(station, scanjob, liveplotwindow=False, verbose=0)
```

Comment:
This test does not check anything. It would be nice to test whether the 'label' value is indeed what you expect for the default case, such that the added function `_dataset_record_label` works fully as expected.